src/plugins/ui/evaluation.ts

```diff
@@ -277,7 +279,7 @@ export class EvaluationPlugin extends UIPlugin {
     ): any | any[][] {
       const range: Range = references[position];
 
-      if (isMeta) {
+      if (isMeta && !range.invalidSheetName) {
```

Comment:
This is not correct as, if `isMeta` is true but the range is on an invalid sheet, the code will continue outside the `if` statement, which could contains code that assume that `isMeta` is false.
It works by chance because there is another check after that which checks that the sheet name is valid.

See my comment above to improve it.